agent/consul/stream/event_publisher_test.go

```diff
@@ -190,16 +194,18 @@ func TestEventPublisher_SubscribeWithIndex0_FromCache(t *testing.T) {
 
 	publisher := NewEventPublisher(newTestSnapshotHandlers(), time.Second)
 	go publisher.Run(ctx)
+
 	sub, err := publisher.Subscribe(req)
 	require.NoError(t, err)
-	sub.Unsubscribe()
+	defer sub.Unsubscribe()
```

Comment:
I'm curious about why this needed to change as it changes the behaviour of the test. Was it a bug before? How did it pass if so?

Reply:
Hmm I think I see. this subscription was just for the side-effect of priming the snapshot cache for the TopicKey. So it doesn't matter if it ends now or later because the real test here is that the second subscription hits the warm cache right?

But now it _must_ end later otherwise the immediate Unsubscribe will garbage collect the buffer and snapshot and so prevent the next subscription from hitting a cache as desired.

Cool!

Reply:
Bingo 🎰 😄